src/LaunchDarkly.EventSource/ConnectStrategy.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using LaunchDarkly.EventSource.Events;
+using LaunchDarkly.EventSource.Exceptions;
+using LaunchDarkly.Logging;
+
+using static System.Net.WebRequestMethods;
+
+namespace LaunchDarkly.EventSource
+{
+    /// <summary>
+    /// An abstraction of how EventSource should obtain an input stream.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The default implementation is <see cref="HttpConnectStrategy"/>, which makes HTTP
+    /// requests. To customize the HTTP behavior, you can use methods of <see cref="HttpConnectStrategy"/>:
+    /// </para>
+    /// <example><code>
+    ///
+    /// </code></example>>
+    /// <para>
+    /// Or, if you want to consume an input stream from some other source, you can
+    /// create your own subclass of ConnectStrategy.
+    /// </para>
+    /// <para>
+    /// Instances of this class should be immutable and not contain any state that
+    /// is specific to one active stream.The {@link ConnectStrategy.Client}
+    /// that they produce is stateful and belongs to a single EventSource.
+    /// </para>
+    /// </remarks>
+    public abstract class ConnectStrategy
```

Comment:
Equivalent to the [Java version](https://github.com/launchdarkly/okhttp-eventsource/blob/main/src/main/java/com/launchdarkly/eventsource/ConnectStrategy.java).